ui/src/views/Provisioners/ClaimedTasks/index.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from 'react';
+import { graphql } from 'react-apollo';
+import dotProp from 'dot-prop-immutable';
+import { TableRow, TableCell, Chip, Box, Typography } from '@material-ui/core';
+import LinkIcon from 'mdi-react/LinkIcon';
+import HourglassIcon from 'mdi-react/HourglassIcon';
+import WorkerIcon from 'mdi-react/WorkerIcon';
+import Spinner from '../../../components/Spinner';
+import Dashboard from '../../../components/Dashboard';
+import { VIEW_WORKER_POOL_PENDING_TASKS_PAGE_SIZE } from '../../../utils/constants';
+import claimedTasks from './claimedTasks.graphql';
+import ConnectionDataTable from '../../../components/ConnectionDataTable';
+import Link from '../../../utils/Link';
+import TableCellItem from '../../../components/TableCellItem';
+import DateDistance from '../../../components/DateDistance';
+import Breadcrumbs from '../../../components/Breadcrumbs';
+
+@graphql(claimedTasks, {
+  options: props => ({
+    variables: {
+      taskQueueId: `${props.match.params.provisionerId}/${props.match.params.workerType}`,
+      connection: {
+        limit: VIEW_WORKER_POOL_PENDING_TASKS_PAGE_SIZE,
+      },
+    },
+  }),
+})
+export default class WMViewClaimedTasks extends Component {
+  handlePageChange = ({ cursor, previousCursor }) => {
+    const {
+      data: { fetchMore },
+      match: { params },
+    } = this.props;
+
+    return fetchMore({
+      query: claimedTasks,
+      variables: {
+        taskQueueId: `${params.provisionerId}/${params.workerType}`,
+        connection: {
+          limit: VIEW_WORKER_POOL_PENDING_TASKS_PAGE_SIZE,
+          cursor,
+          previousCursor,
+        },
+      },
+      updateQuery(previousResult, { fetchMoreResult: { listClaimedTasks } }) {
+        // use dotProp.set to avoid lint warning about assigning to properties
+        return dotProp.set(
+          previousResult,
+          'listClaimedTasks',
+          listClaimedTasks
+        );
+      },
+    });
+  };
+
+  get workersLink() {
+    const { provisionerId, workerType } = this.props.match.params;
+
+    return `/provisioners/${encodeURIComponent(
+      provisionerId
+    )}/worker-types/${encodeURIComponent(workerType)}`;
+  }
+
+  renderRow({ node: { taskId, runId, claimed, task, workerGroup, workerId } }) {
+    return (
+      <TableRow key={taskId}>
+        <TableCell>
+          <Link to={`/tasks/${taskId}`}>
+            <TableCellItem button>
+              {taskId}
+              <LinkIcon size={16} />
+            </TableCellItem>
+          </Link>
+        </TableCell>
+        <TableCell>{runId}</TableCell>
+        <TableCell>{workerGroup}</TableCell>
+        <TableCell>
+          <Link to={`${this.workersLink}/workers/${workerGroup}/${workerId}`}>
```

Comment:
Should `workerGroup` and `workerId` be URI encoded here?

Same question to the other `<Link to={...}>` instances down in this file.

Reply:
good question! 
I don't think it is necessary, there are usages without encoding elsewhere in the code. The only place it matters was for `workerPoolId` since it includes `/` which should be encoded in order to make it easier for routers